tools/lxdclient/client_instance.go

```diff
@@ -308,30 +305,42 @@ func (client *instanceClient) Addresses(name string) ([]network.Address, error)
 
 // AttachDisk attaches a disk to an instance.
 func (client *instanceClient) AttachDisk(instanceName, deviceName string, disk DiskDevice) error {
-	props := []string{"path=" + disk.Path, "source=" + disk.Source}
+	container, _, err := client.raw.GetContainer(instanceName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	container.Devices[deviceName] = map[string]string{
+		"path":   disk.Path,
+		"source": disk.Source,
+	}
 	if disk.Pool != "" {
-		props = append(props, "pool="+disk.Pool)
+		container.Devices[deviceName]["pool"] = disk.Pool
 	}
 	if disk.ReadOnly {
-		props = append(props, "readonly=true")
+		container.Devices[deviceName]["readonly"] = "true"
 	}
-	resp, err := client.raw.ContainerDeviceAdd(instanceName, deviceName, "disk", props)
+	resp, err := client.raw.UpdateContainer(instanceName, container.Writable(), "")
 	if err != nil {
 		return errors.Trace(err)
 	}
-	if err := client.raw.WaitForSuccess(resp.Operation); err != nil {
+	if err := resp.Wait(); err != nil {
 		return errors.Trace(err)
 	}
 	return nil
 }
 
 // RemoveDevice removes a device from an instance.
 func (client *instanceClient) RemoveDevice(instanceName, deviceName string) error {
-	resp, err := client.raw.ContainerDeviceDelete(instanceName, deviceName)
+	container, _, err := client.raw.GetContainer(instanceName)
```

Comment:
another place where we should probably be reading the ETAG and passing it back in.
Possibly in a loop in case the ETAG is changing.